compose/project.py

```diff
@@ -55,23 +58,47 @@ def from_config(cls, name, config_data, client):
         use_networking = (config_data.version and config_data.version >= 2)
         project = cls(name, [], client, use_networking=use_networking)
 
-        if use_networking:
-            remove_links(config_data.services)
+        custom_networks = []
+        if config_data.networks:
+            for network_name, data in config_data.networks.items():
+                custom_networks.append(
+                    Network(
+                        client=client, project=name, name=network_name,
+                        driver=data.get('driver'),
+                        driver_opts=data.get('driver_opts'),
+                        external_name=data.get('external_name'),
+                    )
+                )
 
         for service_dict in config_data.services:
-            links = project.get_links(service_dict)
+            if use_networking:
+                networks = get_networks(
+                    service_dict,
+                    custom_networks + [project.default_network])
+                net = Net(networks[0]) if networks else Net("none")
```

Comment:
I think this means we'll attempt to re-connect to the first network twice?  Maybe that's fine? I don't really know.